### ISSUE TITLE
Nuxt example? I got Element ID doesn't exist

### ISSUE THREAD

**Issue by lsshawn** [botui/botui#58]
I'm getting this error:

> Uncaught (in promise) Error: BotUI: Element with id #sample-bot does not exist.

And here's my code:

`<template>
  <div>
    <h1>BOTUI test</h1>
    <no-ssr>
    <div id="sample-bot">
      <div id="sample-bot">
        <bot-ui></bot-ui>
      </div>
    </div>
    </no-ssr>
  </div>
</template>

<script>
import initBot from '@/botui/sample-bot.js'

export default {
  async mounted () {
    await initBot()
  }
}
</script>
`
Anyone have working example with Vue Nuxt?

**Comment by HassanNemir** [comment]
it's DOM issue, that the class constructor looks for the element before it's mounted, try adding the element to index.html and check this https://vuejs.org/v2/api/#Options-Lifecycle-Hooks

**Comment by Fab1en** [comment]
I have posted an issue saying that botui cannot be used as a component as it is. You have to rework the code to get rid of the need to provide a root component id. See #77

**Comment by Alagaesia93** [comment]
Any news on this? I am currently trying to include BotUI in a nuxt project and nothing seems to work.

**Comment by Alagaesia93** [comment]
Working solution:

```
import vue from 'vue'
import Botui from 'botui'

export default {
  data() {
    return {
      botui: ''
    }
  },
  async mounted() {
    await this.$nextTick()
    this.botui = new Botui('botui', { vue })
  },
  methods: {
    startBot() {
      this.botui.message
        .add({
          content: ({
            text: 'Hello'
          })
        })
      }
    }
}
```

Found on https://discordapp.com/invite/VApZF5W

**Comment by edlgg** [comment]
Does anybody have a complete working example? I cant make it run. Im getting this error 'Unknown custom element: <bot-ui> - did you register the component correctly?'

Component:
```
<template>
  <div>
    <h1>BOTUI test</h1>
    <no-ssr>
      <div id="botui">
        <bot-ui></bot-ui>
      </div>
    </no-ssr>
  </div>
</template>
<script>
import vue from 'vue'
import Botui from 'botui'

export default {
  data() {
    return {
      botui: ''
    }
  },
  async mounted() {
    await this.$nextTick()
    this.botui = new Botui('botui', { vue })
  },
  methods: {
    startBot() {
      this.botui.message.add({
        content: {
          text: 'Hello'
        }
      })
    }
  }
}
</script>

```

**Comment by razorabhu1995** [comment]
@Alagaesia93 @edlgg @lsshawn 

did any one find proper solution for this??

**Comment by Alagaesia93** [comment]
@razorabhu1995 Yes it works for me, as in my previous example. The secret is to wrap bot-ui tag in no-ssr and wait for this.$nextTick() before new Botui. If you need help please report your code!
